data/lib/service/team/team_service.dart

```diff
@@ -27,99 +27,81 @@ class TeamService {
-          name: teamRequestModel.name,
-          name_lowercase: teamRequestModel.name_lowercase,
-          id: teamRequestModel.id,
-          city: teamRequestModel.city,
-          created_at: teamRequestModel.created_at,
-          created_by: teamRequestModel.created_by,
-          profile_img_url: teamRequestModel.profile_img_url,
-          players: member);
-      return team;
+      final team = teamDoc.data();
+
+      final users = await getMemberListFromUserIds(
+        team?.players.map((e) => e.id).toList() ?? [],
+      );
+
+      final players = team?.players.map((player) {
+        final user = users.firstWhere((element) => element.id == player.id);
+        return player.copyWith(user: user);
+      }).toList();
+
+      return team?.copyWith(players: players ?? []) ??
+          deActiveDummyTeamModel(teamId);
     }).handleError((error, stack) => AppError.fromError(error, stack));
   }
 
   Stream<List<TeamModel>> getUserRelatedTeams() {
     if (_currentUserId == null) {
       return Stream.value([]);
     }
-    final filter = Filter.or(
-      Filter(FireStoreConst.createdBy, isEqualTo: _currentUserId),
-      Filter(FireStoreConst.players, arrayContains: _currentUserId),
-    );
-    return _teamsCollection
-        .where(filter)
-        .snapshots()
-        .asyncMap((snapshot) async {
-      return await Future.wait(snapshot.docs.map((mainDoc) async {
-        AddTeamRequestModel teamRequestModel = mainDoc.data();
-
-        final member = (teamRequestModel.players?.isNotEmpty ?? false)
-            ? await getMemberListFromUserIds(teamRequestModel.players ?? [])
-            : null;
-
-        final team = TeamModel(
-          name: teamRequestModel.name,
-          name_lowercase: teamRequestModel.name_lowercase,
-          id: teamRequestModel.id,
-          city: teamRequestModel.city,
-          created_at: teamRequestModel.created_at,
-          created_by: teamRequestModel.created_by,
-          profile_img_url: teamRequestModel.profile_img_url,
-          players: member,
-        );
-        return team;
-      }));
+    return _teamsCollection.snapshots().asyncMap((snapshot) async {
+      final teams = await Future.wait(
+        snapshot.docs.map((mainDoc) async {
+          final team = mainDoc.data();
+          final isUserInTeam =
+              team.players.any((element) => element.id == _currentUserId);
+          if (isUserInTeam) {
+            final users = await getMemberListFromUserIds(
+              team.players.map((e) => e.id).toList(),
+            );
+
+            final players = team.players.map((player) {
+              final user =
+                  users.firstWhere((element) => element.id == player.id);
+              return player.copyWith(user: user);
+            }).toList();
+
+            return team.copyWith(players: players);
+          } else {
+            return null;
+          }
+        }),
+      );
+      return teams.whereType<TeamModel>().toList();
```

Comment:
why do we need whereType?

Reply:
we are returning a null value in team list, we need to remove the null values, which is why we have used whereType.